crates/cli-support/src/js/mod.rs

```diff
@@ -217,26 +213,29 @@ impl<'a> Context<'a> {
 
         let mut shim = String::new();
 
-        shim.push_str("let imports = {};\n");
+        shim.push_str("\nlet imports = {};\n");
 
-        if self.config.mode.nodejs_experimental_modules() {
+        if self.config.mode.uses_es_modules() {
             for (i, module) in imports.iter().enumerate() {
                 if module.as_str() != PLACEHOLDER_MODULE {
                     shim.push_str(&format!("import * as import{} from '{}';\n", i, module));
                 }
             }
-        }
-
-        for (i, module) in imports.iter().enumerate() {
-            if module.as_str() == PLACEHOLDER_MODULE {
-                shim.push_str(&format!(
-                    "imports['{0}'] = module.exports;\n",
-                    PLACEHOLDER_MODULE
-                ));
-            } else if self.config.mode.nodejs_experimental_modules() {
-                shim.push_str(&format!("imports['{}'] = import{};\n", module, i));
-            } else {
-                shim.push_str(&format!("imports['{0}'] = require('{0}');\n", module));
+            for (i, module) in imports.iter().enumerate() {
+                if module.as_str() != PLACEHOLDER_MODULE {
+                    shim.push_str(&format!("imports['{}'] = import{};\n", module, i));
+                }
+            }
```

Comment:
The import steps are added to the index file now, so no circular import, whether that would be an issue or not.

Why it generates an import for `{name}_bg.js` as shown in the snippet in the PR I'm not sure, but it's correct. That's what the import table looks like:

```
Import[2]:
 - func[0] sig=1 <./node_bg.js.__wbindgen_string_new> <- ./node_bg.js.__wbindgen_string_new
 - func[1] sig=2 <./node_bg.js.__wbindgen_throw> <- ./node_bg.js.__wbindgen_throw
```